src/commands/giveaway.ts

```diff
@@ -1,54 +1,88 @@
-import { Message, TextChannel } from "discord.js";
+import { SlashCommandBuilder } from "discord.js";
+import { Command } from "../types/command";
+import { MessageReaction, User } from "discord.js";
 
-export let giveawayActive = false;
-let participants: string[] = [];
-let numberOfWinners = 1;
-let giveawayTimeout: NodeJS.Timeout | null = null;
+export class GiveawayCommand implements Command {
+	name = "giveaway";
+	description = "Creates a giveaway.";
+	slashCommandConfig = new SlashCommandBuilder()
+		.setName(this.name)
+		.setDescription(this.description)
+		.addChannelOption((option) =>
+			option.setName("channel").setDescription("The channel to echo into").setRequired(true)
+		)
+		.addStringOption((option) =>
+			option.setName("duration").setDescription("The duration of the giveaway.").setRequired(true)
+	)
+		.addNumberOption((option) =>
+			option.setName("number_of_winners").setDescription("The number of winners for the giveaway.")
+		);
```

Comment:
Would recommend requiring the user who runs the `/giveaway` command to have Administrator permissions, as we wouldn't want anyone running this command. This can be done by adding `.setDefaultMemberPermissions(PermissionFlagsBits.Administrator);` to `slashCommandConfig`, if you import `PermissionFlagsBits` from `discord.js`.